lib/util/memcache.ex

```diff
@@ -73,6 +73,31 @@ defmodule Antikythera.Memcache do
     end)
   end
 
+  @doc """
+  Try to read a value associated with the `key` from #{inspect(__MODULE__)}.
+
+  `fun` is evaluated if a value is not found, and the new value returned by `fun` is stored in #{inspect(__MODULE__)}.
+  If a value is found in #{inspect(__MODULE__)} or writing the new value to #{inspect(__MODULE__)} succeeds, the value is returned as `{:ok, value}`, but if writing the new value fails, an error is returned in the same manner as `write/5`.
+
+  Parameters `lifetime_in_sec` and `prob_lifetime_ratio` are used to call `write/5` and the details are described above.
+  """
+  defun read(key                 :: Key.t,
+             epool_id            :: v[EPoolId.t],
+             lifetime_in_sec     :: v[non_neg_integer],
+             fun                 :: (-> Value.t),
+             prob_lifetime_ratio :: v[NormalizedFloat.t] \\ @default_ratio) :: R.t(Value.t, :too_large_key | :too_large_value) do
```

Comment:
I think it'd be nicer to flip `fun` and `prob_lifetime_ratio` so that on calling side possibly long `fn` expression comes last in the parameter list.

Reply:
I have misunderstood that middle args can have default values!
Fixed in 330a2b6cefc2d391f3cf4e0b997e2093bc09b0ca